packages/core/src/component.ts

```diff
@@ -14,6 +14,15 @@ export interface ProviderMap {
   [key: string]: Constructor<Provider<BoundValue>>;
 }
 
+/**
+ * A map of name/class pairs for binding static assets
+ */
+export type staticAssetsMap = {
```

Comment:
Uff, then name and comment seems very misleading to me, because there is no map involved AFAICT.

Also please use `interface` instead of `type`.